pytorch_lightning/loops/batch/training_batch_loop.py

```diff
@@ -161,68 +150,37 @@ def _run_optimization(
         self,
         batch_idx: int,
         split_batch: Any,
-        opt_idx: Optional[int] = None,
-        optimizer: Optional[torch.optim.Optimizer] = None,
     ) -> Optional[ClosureResult]:
         """Runs closure (train step + backward) together with optimization if necessary.
 
         Args:
             batch_idx: the index of the current batch
             split_batch: the current tbptt split of the whole batch
-            opt_idx: the index of the current optimizer or `None` in case of manual optimization
-            optimizer: the current optimizer or `None` in case of manual optimization
         """
-        # toggle model params
-        self._run_optimization_start(opt_idx, optimizer)
-
-        closure = self._make_closure(split_batch, batch_idx, opt_idx, optimizer, self._hiddens)
-
-        if self.trainer.fit_loop.should_accumulate():
-            # For gradient accumulation
-
-            # -------------------
-            # calculate loss (train step + train step end)
-            # -------------------
-            # automatic_optimization: perform ddp sync only when performing optimizer_step
-            with _block_parallel_sync_behavior(self._trainer):
-                closure()
-
-        # ------------------------------
-        # BACKWARD PASS
-        # ------------------------------
-        # gradient update with accumulated gradients
-        else:
-            if self.trainer.lightning_module.automatic_optimization:
-                self._optimizer_step(optimizer, opt_idx, batch_idx, closure)
-            else:
-                closure()
-
+        closure = self._make_closure(split_batch, batch_idx, self._hiddens)
+        closure()
```

Comment:
eliminating the closure here for manual optimization will be a direct follow up to this PR

Reply:
Add a TODO then :)